src/components/molecules/CreateVenueModal/CreateVenueModal.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useCallback, useState } from "react";
+import { Form, Modal } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+
+import { createUrlSafeName, createVenue_v2 } from "api/admin";
+
+import { useUser } from "hooks/useUser";
+import { useVenueId } from "hooks/useVenueId";
+
+import { venueLandingUrl } from "utils/url";
+
+import { VenueTemplate } from "types/venues";
+
+import { newVenueSchema } from "pages/Admin/Details/ValidationSchema";
+
+import "./CreateVenueModal.scss";
+
+interface CreateVenueModalProps {
+  isVisible: boolean;
+  onHide?: () => void;
+}
+
+export interface FormValues {
+  name: string;
+}
+
+export const CreateVenueModal: React.FC<CreateVenueModalProps> = ({
+  isVisible,
+  onHide,
+}) => {
+  const [isLoading, setLoading] = useState<boolean>(false);
+  const venueId = useVenueId();
+  const { user } = useUser();
+
+  const hide = useCallback(() => {
+    onHide && onHide();
+  }, [onHide]);
+
+  const onSubmit = useCallback(
+    async (vals: FormValues) => {
+      if (!user || isLoading) return;
+      setLoading(true);
+
+      try {
+        const venue = {
+          name: vals.name,
+          template: VenueTemplate.partymap,
+        };
+        await createVenue_v2(venue, user);
+        hide();
+      } catch (e) {
+        console.error(e);
+      } finally {
+        setLoading(false);
+      }
+    },
+    [hide, isLoading, user]
+  );
+
+  const { watch, register, errors, handleSubmit } = useForm<FormValues>({
+    mode: "onSubmit",
+    reValidateMode: "onSubmit",
+    validationSchema: newVenueSchema,
+    validationContext: {
+      editing: !!venueId,
+    },
+  });
+
+  const values = watch();
+
+  const urlSafeName = values.name
```

Comment:
Presumably we should extract this into a helper function?

Also, we can construct the URL in a better/safer way than string concatenation like this. I believe we already do similar in `utils/url` using the `new Url` and similar.

Reply:
Looks like nothing of the already existing code works for this case. I just created a new function, you can take a look in the last commit.

Reply:
https://github.com/sparkletown/sparkle/pull/1351/commits/5b605b866c48d16a4be9f814684e8b2b6992493f

Reply:
I didn't expect one of the functions there would have worked, but I was talking about the method we use to construct a URL, not using string concatenation, but using `new URL`:

- https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

Looking at your function, it will probably end up looking something like this:

```typescript
export const generateVenueLandingUrl = (value?: string): string | undefined => {
  if (!value) return undefined

  const path = venueLandingUrl(createUrlSafeName(value))

  return new URL(path, window.location.host).toString()
```

It would be good to have a more contextually relevant name than `value` here too if possible. Is it intended to be the `venueId` or `venueName` or similar? If so, can we call it that?